api/app/tests/apps/api/v1/test_private_signal_endpoint.py

```diff
@@ -1975,6 +1976,125 @@ def test_only_for_visible_parent(self):
+        categories, then the "can_view_signal" should all be True
+        """
+        parent_category = ParentCategoryFactory.create()
+        child_category_1 = CategoryFactory.create(parent=parent_category)
+        child_category_2 = CategoryFactory.create(parent=parent_category)
+
+        department = DepartmentFactory.create()
+
+        CategoryDepartmentFactory.create(category=child_category_1, department=department, can_view=True)
+        CategoryDepartmentFactory.create(category=child_category_2, department=department, can_view=True)
+
+        self.sia_read_write_user.profile.departments.add(department)
+        self.client.force_authenticate(user=self.sia_read_write_user)
+
+        parent_signal = SignalFactory.create(category_assignment__category=child_category_1)
+        SignalFactory.create(parent=parent_signal, category_assignment__category=child_category_1)
+        SignalFactory.create(parent=parent_signal, category_assignment__category=child_category_2)
+
+        response = self.client.get(self.child_endpoint.format(pk=parent_signal.pk))
+        self.assertEqual(response.status_code, 200)
+
+        response_json = response.json()
+        self.assertEqual(response_json['count'], 2)
+        self.assertTrue(response_json['results'][0]['can_view_signal'])
+        self.assertTrue(response_json['results'][1]['can_view_signal'])
+
+    def test_shows_children_can_view_all_children_false(self):
+        """
+        User has no permissions to view all Signals in certain categories, and all children belong to one of those
+        categories, then the "can_view_signal" should all be False
+        """
+        parent_category = ParentCategoryFactory.create()
+        child_category_1 = CategoryFactory.create(parent=parent_category)
+        child_category_2 = CategoryFactory.create(parent=parent_category)
+
+        department = DepartmentFactory.create()
+        CategoryDepartmentFactory.create(category=child_category_1, department=department, can_view=True)
+
+        self.sia_read_write_user.profile.departments.add(department)
+        self.client.force_authenticate(user=self.sia_read_write_user)
+
+        parent_signal = SignalFactory.create(category_assignment__category=child_category_1)
+        SignalFactory.create_batch(2, parent=parent_signal, category_assignment__category=child_category_2)
+
+        response = self.client.get(self.child_endpoint.format(pk=parent_signal.pk))
+        self.assertEqual(response.status_code, 200)
+
+        response_json = response.json()
+        self.assertEqual(response_json['count'], 2)
+        self.assertFalse(response_json['results'][0]['can_view_signal'])
+        self.assertFalse(response_json['results'][1]['can_view_signal'])
+
+    def test_shows_children_can_view_all_children_mixed(self):
+        """
+        User has permissions to view all Signals in certain categories, and all not ll children belong to one of those
```

Comment:
`, and all not ll children belong to one of those` rephrase

Reply:
Rephrased